r/src/arrow_types.h

```diff
@@ -0,0 +1,150 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <RcppCommon.h>
+
+#undef Free
+#include <arrow/api.h>
+#include <arrow/type.h>
+
+#define R_ERROR_NOT_OK(s) do { if(!s.ok()) Rcpp::stop(s.ToString()); } while (0);
+
+template <typename T>
+class static_ptr {
```

Comment:
What is the purpose of this? Is this new code?

Reply:
Yeah, I'll do it differently. This is about not `delete`ing the default memory pool, so having a smart pointer class that leaves alone its pointer. I'll clean that up.

Reply:
Updated it so that it also uses `std::shared_ptr` with a noop deleter. 

```
std::shared_ptr<arrow::MemoryPool> MemoryPool_default(){
  return std::shared_ptr<arrow::MemoryPool>(arrow::default_memory_pool(), NoDelete<arrow::MemoryPool>());
}
```